src/global_scheduler/global_scheduler_algorithm.c

```diff
@@ -24,6 +24,7 @@ void handle_local_scheduler_heartbeat(global_scheduler_state *state) {
 
 void handle_new_local_scheduler(global_scheduler_state *state,
                                 db_client_id db_client_id) {
-  local_scheduler local_scheduler = {.id = db_client_id};
+  local_scheduler local_scheduler;
+  local_scheduler.id = db_client_id;
```

Comment:
Sorry to pick on this again, but you can actually just shorten this to `local_scheduler local_scheduler = {db_client_id};`.
Also, be aware that there's a slight difference between your new version and using brace initialization -- if any fields are added later, they'll be zero-initialized when you use brace initialization, but they'll be uninitialized in your version. So if you're trying to initialize the whole struct, use braces.